pkcs7/src/signed_data_content.rs

```diff
@@ -0,0 +1,54 @@
+//! `signed-data` content type [RFC 5652 § 5](https://datatracker.ietf.org/doc/html/rfc5652#section-5)
+
+use crate::{signer_info::{SignerInfos}, encapsulated_content_info::EncapsulatedContentInfo, certificate_choices::CertificateChoices, revocation_info_choices::RevocationInfoChoices, cms_version::CMSVersion};
+use der::{Sequence, asn1::{SetOfVec}};
+use spki::{AlgorithmIdentifierRef};
+
+/// ```text
+/// DigestAlgorithmIdentifier ::= AlgorithmIdentifier
+/// ```
+type DigestAlgorithmIdentifier<'a> = AlgorithmIdentifierRef<'a>;
+
+/// ```text
+/// DigestAlgorithmIdentifiers ::= SET OF DigestAlgorithmIdentifier
+/// ```
+type DigestAlgorithmIdentifiers<'a> = SetOfVec<DigestAlgorithmIdentifier<'a>>;
```

Comment:
Maybe I did something weird locally, but I'm surprised this isn't complaining about the lack of the `alloc` feature from the `der` crate. Did you build with all features enabled locally?

Reply:
Now that you say it I'm wondering as well. I ran everything without additional features enabled.

Reply:
Yeah, it built for me locally without additional features as well, strange.

Reply:
You're pulling in the `x509-cert` crate which unconditionally activates the `alloc` feature, and feature unification means it's activated here too.

It would probably be good to factor the relevant parts into a common crate which can be shared. We had the `x501` crate for this purpose at one point, although I'm not sure that's quite what you need.

Reply:
Thanks for explaining!

If there *is* interest in maintaining a non-`alloc` part of `pkcs7`, this is how I did it in my own changes:

```toml
[features]
alloc = ["der/alloc"]
std = ["alloc"]
signed-data = ["alloc", "dep:x509-cert"]
```

(with `x509-cert` then being marked as optional.)

Reply:
I'm guessing that, like #765, this crate should probably move to all owned types and make `alloc` a hard dependency, since heapless support probably won't be practical for real-world use cases

Reply:
What should we do for now? I don't want to mix this PR with the larger effort of doing something like #765 for this crate as well.

Reply:
I'm +1 on doing the owned refactor in a follow-up (I can help with that) to keep the diff small 🙂